src/Slackbot.Net.Extensions.FplBot/Helpers/SlackWorkSpacePublisher.cs

```diff
@@ -32,15 +33,16 @@ public async Task PublishToAllWorkspaces(string msg)
             }
         }
 
-        public async Task PublishUsingToken(string token, params string[] messages)
+        public async Task PublishUsingToken(string token, params Func<User[], string>[] messages)
         {
             var setup = await _teamRepo.GetSetupByToken(token);
             var slackClient = _slackClientBuilder.Build(token);
+            var users = await slackClient.UsersList();
```

Comment:
Not all messages sent to slack care about slack users (like "Gameweek started!" etc), so I think it's better if we moved this call to part of message formatting so it's only called for those messages that care about slack handles.